builtin/providers/aws/resource_aws_codecommit_repository.go

```diff
@@ -0,0 +1,185 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/codecommit"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCodeCommitRepository() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCodeCommitRepositoryCreate,
+		Update: resourceAwsCodeCommitRepositoryUpdate,
+		Read:   resourceAwsCodeCommitRepositoryRead,
+		Delete: resourceAwsCodeCommitRepositoryDelete,
+
+		Schema: map[string]*schema.Schema{
+			"repository_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if len(value) > 100 {
+						errors = append(errors, fmt.Errorf(
+							"%q cannot be longer than 100 characters", k))
+					}
+					return
+				},
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if len(value) > 1000 {
+						errors = append(errors, fmt.Errorf(
+							"%q cannot be longer than 1000 characters", k))
+					}
+					return
+				},
+			},
+
+			"arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"repository_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"clone_url_http": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"clone_url_ssh": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"default_branch": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCodeCommitRepositoryCreate(d *schema.ResourceData, meta interface{}) error {
+	codecommitconn := meta.(*AWSClient).codecommitconn
```

Comment:
Stylistically we've tended to call the client just `conn` when it's a local variable.